src/plugins/ui_actions/public/cell_actions/components/cell_actions_context.tsx

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import React, { createContext, FC } from 'react';
+import type { Action } from '../../actions';
+
+type GetActionsType = undefined | ((trigger: string, context: object) => Promise<Action[]>);
+
+const initialContext = {
+  getCompatibleActions: undefined,
+};
+
+export const CellActionsContext = createContext<{ getCompatibleActions: GetActionsType }>(
+  initialContext
+);
+
+interface CellActionsContextProviderProps {
+  /**
```

Comment:
this could use more explanation, what does compatible means? what does the getTriggerCompatibleActions function do?

Reply:
That returns a list of actions that were attached to the specified trigger and which should still be visible given the value in the `context`.

I guess, in a React component, the naming could be simplified to something like `getActions`.

Reply:
The `uiActions` plugin exposes two functions to get the actions registered to a trigger: `getTriggerActions` and `getTriggerCompatibleActions`. The first gets all the functions registered, and the latter does the same but it also filters the actions by calling the `isCompatible` method of every Actions, which is what we need here. 

I think is important to name this prop with the _compatible_ word in it, to prevent passing the wrong function. 
And, in any case, I would rename the prop to the same `getTriggerCompatibleActions` name the plugin exports, to make it even more obvious.